### PR TITLE
Problem: autosync unlock request doesn't check if the wallet was already added to auto sync (cro-395)

### PR DESCRIPTION
**Pull request by leejw51crypto**
make inserting wallet as upsert to prevent adding redundant wallets

### CODE REVIEW

File: client-index/src/auto_sync_core.rs [hunk 1]
```diff
@@ -217,7 +217,16 @@ where
     pub fn get_queue(&self) -> Sender<OwnedMessage> {
         self.my_sender.clone()
     }
-
+    fn update_wallet(&mut self, info: &WalletInfo) -> Result<()> {
+        for i in 0..self.wallets.len() {
```
Comment (devashishdxt) [inline client-index/src/auto_sync_core.rs]: Instead of checking duplicates in a `Vec`. Can we have a `HashSet` or `BTreeSet` of `WalletInfo`?

Comment (leejw51crypto) [reply client-index/src/auto_sync_core.rs]: order is important, so need to use vec or list

Comment (devashishdxt) [reply client-index/src/auto_sync_core.rs]: Why is order important?

**Comment by leejw51crypto** [comment]
agree, i'll change to btreemap

File: client-index/src/auto_sync_core.rs (hunk 1 above)
Comment (leejw51crypto) [reply client-index/src/auto_sync_core.rs]: yes , BTreeMap will be good

**Comment by leejw51crypto** [comment]
sorry, it will work now

**Pull request closed without merging**